### PR TITLE
Delete sessions for stale app instances

### PR DESCRIPTION
**Pull request by garamond**
Fixes https://github.com/orbiting/app/issues/161

### CODE REVIEW

File: packages/notifications/lib/utils.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const deleteSessionForDevices = async (tokens, pgdb) => {
+  return pgdb.query(`
+    DELETE FROM sessions s WHERE s.id IN (
+                                         SELECT d."sessionId" FROM devices d WHERE d.id IN (:tokens)
```
Comment (patte) [inline packages/notifications/lib/utils.js]:
```suggestion
      SELECT d."sessionId" FROM devices d WHERE d.id IN (:tokens)
```

File: packages/notifications/lib/utils.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+                                         )
```
Comment (patte) [inline packages/notifications/lib/utils.js]:
```suggestion
    )
```

File: packages/notifications/lib/utils.js (hunk 1 above)
Comment (patte) [inline packages/notifications/lib/utils.js]: I wonder if/why `IN (:tokens)` works. Pogi says you should use `&&` here: https://pogi.readthedocs.io/en/latest/pitfalls/#named-parameter-column-in-expression

Comment (garamond) [reply packages/notifications/lib/utils.js]: Maybe because of the parens around `:tokens`?

File: packages/notifications/lib/apn.js [hunk 3]
```diff
@@ -58,12 +60,22 @@ const publish = async (args) => {
...
     })
-    debug('sending %d notifications...', tokens.length)
-    return Promise.all(
+    const result = await Promise.all(
       tokens.map(token =>
         provider.send(message, token)
       )
     )
+    debug('APN: #recipients %d, message: %O, result: %O', tokens.length, message, result)
+    const staleTokens = result.failed.reduce((acc, cur) => {
```
Comment (tpreusse) [inline packages/notifications/lib/apn.js]: where does `result.failed` come from? AFAIK `Promise.all` does not return such a thing.

File: packages/notifications/lib/utils.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const deleteSessionForDevices = async (tokens, pgdb) => {
+  return pgdb.query(`
+    DELETE FROM sessions s WHERE s.id IN (
+      SELECT d."sessionId" FROM devices d WHERE ARRAY[d.id] && :tokens
```
Comment (tpreusse) [inline packages/notifications/lib/utils.js]: should this really be `d.id`? Tokens are saved in the `device.token` column.

File: packages/notifications/lib/apn.js (hunk 3 above)
Comment (garamond) [reply packages/notifications/lib/apn.js]: That's true, I have to reduce the `result` array first…

File: packages/notifications/lib/utils.js (hunk 4 above)
Comment (garamond) [reply packages/notifications/lib/utils.js]: 👍

**Pull request merged**